### PR TITLE
feat(ui): display partially received Flux responses

### PR DESCRIPTION
**Pull request by chnn**
A basic fix for #14511. If a Flux response is truncated because it is too large, the partial result will still display as well as a notification.

Screen Shot 2019-07-31 at 4 06 39 PM

The new limits for Flux responses are:

- OSS: 100 MiB
- Cloud 2: 10 MiB

Next we should improve the UX for this event (add suggestions to reduce response size, add button to run query anyway).

I refactored some code along the way (part of #14482).

I also tweaked the first load experience for the dashboard. Before:

before

After:

after

### CODE REVIEW

**Comment by chnn** [comment]
> I'm not familiar with TypeScript, so I don't know that I have much of value to say for this PR. Is there a particular part of this that you want me to look at?

I can't speak for kelvin (who requested the view), but I appreciate you taking a look.

> I see there are AST structures in there, are they actually being used?

Yeah, they were in use before. This PR is just shifting around where the type definitions for the various AST nodes are stored. We use them to do things like analyze the duration of a query, or pick out what buckets are read from or written to in a query.

File: ui/src/shared/apis/query.ts [hunk 1]
```diff
@@ -1,19 +1,177 @@
...
+  }
+
+  const body: Query = {
+    query,
     extern,
-    limitChars: MAX_RESPONSE_CHARS,
+    dialect: {annotations: ['group', 'datatype', 'default']},
+  }
+
+  const controller = new AbortController()
```
Comment (121watts) [inline ui/src/shared/apis/query.ts]: Might want a polyfill for this.  Especially for our Safari brethren.

File: ui/src/shared/utils/insertPreambleInScript.ts [hunk 2]
```diff
@@ -15,10 +15,16 @@ export const insertPreambleInScript = async (
     return `${preamble}\n\n${script}`
   }
 
-  // TODO: replace this with `import {parse} from '@influxdata/flux-parser'`
-  // when the flux team adds location.source data to the rust implementation
+  // TODO: Replace this with `import {parse} from '@influxdata/flux-parser'`
+  // when the Flux team adds location.source data to the rust implementation
```
Comment (121watts) [inline ui/src/shared/utils/insertPreambleInScript.ts]: How dare you.

File: ui/src/types/queries.ts [hunk 3]
```diff
@@ -1,86 +1,4 @@
-import {Source} from 'src/types'
```
Comment (121watts) [inline ui/src/types/queries.ts]: Heck yea!

**Pull request merged**